Missing example for using the AppenderBuilder

Im having trouble building the config for rust4rs programmatically (without a config-file).
From what i gather the problem seems to be that the build()-function will not accept the `&mut AppenderBuilder` returned.

```
let logstash = log4rs_logstash::appender::AppenderBuilder::default()
    .with_hostname("my-hostname")
    .with_port(5000)
    .with_buffer_size(100)
    .with_buffer_lifetime(Duration::from_secs(1))
    .build().unwrap();
```

Give the error:

```
error[E0507]: cannot move out of a mutable reference
   --> testing_log4rs/src/main.rs:73:20
    |
73  |       let logstash = log4rs_logstash::appender::AppenderBuilder::default()
    |  ____________________^
74  | |     .with_hostname("my-hostname")
75  | |     .with_port(5000)
76  | |     .with_buffer_size(100)
77  | |     .with_buffer_lifetime(Duration::from_secs(1))
78  | |     .build().unwrap();
    | |      ------^
    | |______|_____|
    |        |     move occurs because value has type `log4rs_logstash::appender::AppenderBuilder`, which does not implement the `Copy` trait
    |        value moved due to this method call
```
How is this supposed to be used?
Best regards!

--- a/log4rs-logstash/examples/builder.rs
+++ b/log4rs-logstash/examples/builder.rs
@@ -1,0 +1,62 @@
+#[macro_use]
+extern crate log;
+
+use std::time::Duration;
+
+use anyhow::Result as AnyResult;
+use log::LevelFilter;
+use log4rs::{
+    config::{Appender, Root},
+    init_config, Config,
+};
+use signal_hook::{
+    consts::{SIGINT, SIGTERM},
+    iterator::Signals,
+    low_level::exit,
+};
+
+fn main() -> AnyResult<()> {
+    let logstash = log4rs_logstash::appender::AppenderBuilder::default()
+        .with_hostname("my-hostname")
+        .with_port(5000)
+        .with_buffer_size(100)
+        .with_buffer_lifetime(Duration::from_secs(1))
+        .build()
+        .unwrap();
+    let appender = Appender::builder().build("logstash", Box::new(logstash));
+    let config = Config::builder().appender(appender).build(
+        Root::builder()
+            .appender("logstash")
+            .build(LevelFilter::Warn),
+    )?;
+    init_config(config)?;
+    spawn_signal_handler()?;
+
+    loop {
+        std::thread::sleep(Duration::from_secs(1));
+        debug!("Debug");
+        trace!("Trace");
+        info!("Info");
+        warn!("Warn");
+        error!("Error");
+    }
+}
+
+fn spawn_signal_handler() -> AnyResult<()> {
+    let mut signals = Signals::new(&[SIGINT, SIGTERM])?;
+
+    std::thread::spawn(move || {
+        let mut stop_in_progress = false;
+        for _sig in signals.forever() {
+            std::thread::spawn(move || {
+                log::logger().flush();
+                exit(0)
+            });
+            if stop_in_progress {
+                exit(1);
+            }
+            stop_in_progress = true;
+        }
+    });
+    Ok(())
+}

--- a/log4rs-logstash/src/appender.rs
+++ b/log4rs-logstash/src/appender.rs
@@ -51,19 +51,19 @@ impl Default for AppenderBuilder {
 
 impl AppenderBuilder {
     /// Sets threshold for this logger to level.
-    pub fn with_ignore_buffer_level(&mut self, level: LogLevel) -> &mut AppenderBuilder {
+    pub fn with_ignore_buffer_level(mut self, level: LogLevel) -> AppenderBuilder {
         self.ignore_buffer = level;
         self
     }
 
     /// Sets the hostname of the remote server.
-    pub fn with_hostname(&mut self, hostname: impl Into<String>) -> &mut AppenderBuilder {
+    pub fn with_hostname(mut self, hostname: impl Into<String>) -> AppenderBuilder {
         self.hostname = hostname.into();
         self
     }
 
     /// Sets the port of the remote server.
-    pub fn with_port(&mut self, port: u16) -> &mut AppenderBuilder {
+    pub fn with_port(mut self, port: u16) -> AppenderBuilder {
         self.port = port;
         self
     }
@@ -71,7 +71,7 @@ impl AppenderBuilder {
     /// Sets the upperbound limit on the number of records that can be placed in the buffer, once
     /// this size has been reached, the buffer will be sent to the remote server.
     /// If buffer size is 0 or 1 then buffer is not used
-    pub fn with_buffer_size(&mut self, buffer_size: usize) -> &mut AppenderBuilder {
+    pub fn with_buffer_size(mut self, buffer_size: usize) -> AppenderBuilder {
         if buffer_size < 2 {
             self.buffer_size = None;
         } else {
@@ -81,34 +81,31 @@ impl AppenderBuilder {
     }
 
     /// Sets the maximum lifetime of the buffer before send it to the remote server.
-    pub fn with_buffer_lifetime(&mut self, buffer_duration: Duration) -> &mut AppenderBuilder {
+    pub fn with_buffer_lifetime(mut self, buffer_duration: Duration) -> AppenderBuilder {
         self.buffer_lifetime = Some(buffer_duration);
         self
     }
 
     /// Sets the timeout for network connections.
-    pub fn with_connection_timeout(&mut self, timeout: Duration) -> &mut AppenderBuilder {
+    pub fn with_connection_timeout(mut self, timeout: Duration) -> AppenderBuilder {
         self.connection_timeout = Some(timeout);
         self
     }
 
     /// Use tls connection.
-    pub fn with_use_tls(&mut self, use_tls: bool) -> &mut AppenderBuilder {
+    pub fn with_use_tls(mut self, use_tls: bool) -> AppenderBuilder {
         self.use_tls = use_tls;
         self
     }
 
     /// Print period for internal logstash errors.
-    pub fn with_error_period(&mut self, error_period: Duration) -> &mut AppenderBuilder {
+    pub fn with_error_period(mut self, error_period: Duration) -> AppenderBuilder {
         self.error_period = error_period;
         self
     }
 
     /// Additional fields to send to logstash
-    pub fn with_extra_fields(
-        &mut self,
-        extra_fields: HashMap<String, Value>,
-    ) -> &mut AppenderBuilder {
+    pub fn with_extra_fields(mut self, extra_fields: HashMap<String, Value>) -> AppenderBuilder {
         self.extra_fields = extra_fields;
         self
     }

--- a/log4rs-logstash/src/config.rs
+++ b/log4rs-logstash/src/config.rs
@@ -39,26 +39,26 @@ impl Deserialize for AppenderDeserializer {
         _deserializers: &Deserializers,
     ) -> AnyResult<Box<Self::Trait>> {
         let mut builder = AppenderBuilder::default();
-        builder
+        builder = builder
             .with_hostname(&config.hostname)
             .with_port(config.port)
             .with_use_tls(config.use_tls.unwrap_or(false));
         if let Some(buffer_size) = config.buffer_size {
-            builder.with_buffer_size(buffer_size);
+            builder = builder.with_buffer_size(buffer_size);
         }
         if let Some(buffer_lifetime) = config.buffer_lifetime {
-            builder.with_buffer_lifetime(buffer_lifetime);
+            builder = builder.with_buffer_lifetime(buffer_lifetime);
         }
         if let Some(connection_timeout) = config.connection_timeout {
-            builder.with_connection_timeout(connection_timeout);
+            builder = builder.with_connection_timeout(connection_timeout);
         }
         if let Some(ignore_level) = config.ignore_buffer_level {
-            builder.with_ignore_buffer_level(ignore_level);
+            builder = builder.with_ignore_buffer_level(ignore_level);
         }
         if let Some(error_period) = config.error_period {
-            builder.with_error_period(error_period);
+            builder = builder.with_error_period(error_period);
         }
-        builder.with_extra_fields(config.extra_fields.unwrap_or_default());
+        builder = builder.with_extra_fields(config.extra_fields.unwrap_or_default());
         let appender = builder.build()?;
 
         Ok(Box::new(appender))

